src/trajectory.py

```diff
@@ -0,0 +1,110 @@
+import pickle
+from pathlib import Path
+from typing import Optional
+
+from pymatgen.core import Lattice
+from pymatgen.core.trajectory import Trajectory as PymatgenTrajectory
+from pymatgen.io import vasp
+
+
+class Trajectory(PymatgenTrajectory):
+
+    @classmethod
+    def from_cache(cls, cache: str | Path):
+        """Load data from cache using pickle.
+
+        Parameters
+        ----------
+        cache : Path
+            Name of cache file
+        """
+        with open(cache, 'rb') as f:
+            obj = pickle.load(f)
+        return obj
+
+    def to_cache(self, cache: str | Path):
+        """Dump data to cache using pickle.
+
+        Parameters
+        ----------
+        cache : Path
+            Name of cache file
+        """
+        with open(cache, 'wb') as f:
+            pickle.dump(self, f)
+
+    @classmethod
+    def from_vasprun(cls,
+                     xml_file: str | Path,
+                     cache: Optional[str | Path] = None):
+        """Load data from vasprun.xml.
+
+        Parameters
+        ----------
+        xml_file : Path
+            Path to vasprun.xml
+        cache : Optional[Path], optional
+            Path to cache data for vasprun.xml
+
+        Returns
+        -------
+        data : Data
+            Dataclass with simulation data
+        """
+        if not cache:
+            cache = Path(str(xml_file) + '.cache')
+
+        if Path(cache).exists():
+            try:
+                return cls.from_cache(cache)
+            except Exception as e:
+                print(e)
+                print('Error reading from cache, reading full VaspRun')
+
+        run = vasp.Vasprun(
+            xml_file,
+            parse_dos=False,
+            parse_eigen=False,
+            parse_projected_eigen=False,
+            parse_potcar_file=False,
+        )
+
+        obj = cls.from_structures(
+            run.structures,
+            constant_lattice=True,
+            time_step=run.parameters['POTIM'] * 1e-15,
+        )
+        obj.to_positions()
+        obj.temperature = run.parameters['TEBEG']
+
+        if cache:
+            obj.to_cache(cache)
+
+        return obj
+
+    def get_lattice(self, idx: int | None = None) -> Lattice:
+        """Get lattice.
+
+        Parameters
+        ----------
+        idx : int | None, optional
+            Optionally, get lattice at specified index if the lattice is not constant
+
+        Returns
+        -------
+        lattice : Lattice
+            Pymatgen Lattice object
+        """
+        if self.constant_lattice:
+            return Lattice(self.lattice)
+
+        latt = self.lattices[idx]
+        return Lattice(latt)
+
+    def __getitem__(self, frames):
+        """Hack around pymatgen Trajectory limitations."""
+        new = super().__getitem__(frames)
+        if isinstance(new, PymatgenTrajectory):
+            new.__class__ = self.__class__
+            new.temperature = self.temperature
```

Comment:
shouldn't we create a conversion function here? a `from_PymatgenTrajectory`, that can also keep track of all the changes we introduce.

Reply:
The changes are really small, and if we start using convesion functions we lose the benefit of subclassing.

Reply:
I don't this this is true? the subclassing stil exists, its just the changes are contained in a function, which actually makes it easier to port afterwards, i think.

Reply:
Ah, I see, that makes sense.